javarosa/src/main/java/org/javarosa/core/util/externalizable/ExtUtil.java

```diff
@@ -127,7 +129,14 @@ public static void writeBool(DataOutputStream out, boolean val) throws IOExcepti
     }
 
     public static void writeString(DataOutputStream out, String val) throws IOException {
-        out.writeUTF(val);
+        if (val.getBytes("UTF-8").length > MAX_UNSIGNED_SHORT_VALUE) {

```

Comment:
I'm wondering if this line has bad memory overhead. Is there a way to count the bytes in a string without allocating the whole string into memory again, which is what this is doing, right?